kura/org.eclipse.kura.api/src/main/java/org/eclipse/kura/container/orchestration/ImageConfiguration.java

```diff
@@ -0,0 +1,145 @@
+    }
+
+    /***
+     * Returns a Image's name as a String.
+     * 
+     * @return
+     */
+    public String getImageName() {
+        return this.imageName;
+    }
+
+    /***
+     * Returns a Image's tag as a String.
+     * 
+     * @return
+     */
+    public String getImageTag() {
+        return this.imageTag;
+    }
+
+    /***
+     * Returns a Image's download timeout time as a int.
+     * 
+     * @return
+     */
+    public int getimageDownloadTimeoutSeconds() {
+        return this.imageDownloadTimeoutSeconds;
+    }
+
+    /**
+     * Returns the Registry credentials
+     *
+     * @return
+     */
+    public Optional<RegistryCredentials> getRegistryCredentials() {
+        return this.registryCredentials;
+    }
+
+    /**
+     * Creates a builder for creating a new {@link ImageConfiguration} instance.
+     *
+     * @return the builder.
+     */
+    public static ImageConfigurationBuilder builder() {
+        return new ImageConfigurationBuilder();
+    }
+
+    @Override
+    public int hashCode() {
+        return Objects.hash(this.imageName, this.imageTag, this.registryCredentials);
+    }
+
+    @Override
+    public boolean equals(Object obj) {
+        if (this == obj) {
+            return true;
+        }
+        if (!(obj instanceof ImageConfiguration)) {
+            return false;
+        }
+        ImageConfiguration other = (ImageConfiguration) obj;
+        return Objects.equals(this.imageName, other.imageName) && Objects.equals(this.imageTag, other.imageTag)
+                && Objects.equals(this.registryCredentials, other.registryCredentials);
+    }
+
+    public static final class ImageConfigurationBuilder {
+
+        private String imageName;
+        private String imageTag;
+        private Optional<RegistryCredentials> registryCredentials;
+        private int imageDownloadTimeoutSeconds = 500;
+
+        public ImageConfigurationBuilder setImageName(String imageName) {
+            this.imageName = imageName;
+            return this;
+        }
+
+        public ImageConfigurationBuilder setImageTag(String imageTag) {
+            this.imageTag = imageTag;
+            return this;
+        }
+
+        public ImageConfigurationBuilder setRegistryCredentials(Optional<RegistryCredentials> registryCredentials) {
+            this.registryCredentials = registryCredentials;
+            return this;
+        }
+
+        public ImageConfigurationBuilder setImageDownloadTimeoutSeconds(int imageDownloadTimeoutSeconds) {
+            this.imageDownloadTimeoutSeconds = imageDownloadTimeoutSeconds;
+            return this;
+        }
+
+        public ImageConfiguration build() {
+            ImageConfiguration result = new ImageConfiguration();
+
+            result.imageName = requireNonNull(this.imageName, "Request Image Name cannot be null");
+            result.imageTag = this.imageTag;
+            result.registryCredentials = requireNonNull(this.registryCredentials,
```

Comment:
Do we need explicit credentials? If we have the case where no credentials are needed, can't we set this as empty by default and the the user overrides, when needed, with explicit credentials?
In other words, can we initialise the this.registryCredentials with empty?